### PR TITLE
feat: CWC-756 Opt-out and Unsubscribe Documentation

### PR DESCRIPTION
**Pull request by aosmawhispir**
Committed by Whispir-Softwares on behalf of aosma@whispir.com

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -463,6 +464,61 @@ paths:
...
+        In this sample we will send a message with an opt out option property
+        {
+            "to" : "jsmith@test.com",
+            "subject" : "Test Opt-out and Unsubscribe",
+            "email" : {
+                "body" : "This is a test message",
+                "type" : "text/plain"
+                "optOutOption": {
+                  "rmsUnsubscribeUrl": "@@web_link@@#unsubscribe",
+                  "oneClickApiMapping": "unsubscribe"
```
Comment (lnachor-whispir) [inline openapi.yaml:492]: can you please update the oneClickApiMapping value to a sample generated api mapping id, something like `2ab64862`

File: openapi.yaml [hunk 2]
```diff
@@ -11394,6 +11451,25 @@ components:
...
+        - Email
+      properties:
+        rmsUnsubscribeUrl:
+          type: string
+          example: "@@web_link@@#unsubscribe"
+          description: This is the unsubscribe URL of your message.
+        oneClickApiMapping:
+          type: string
+          example: "unsubscribe"
+          description: This is the API mapping code or ID.
```
Comment (lnachor-whispir) [inline openapi.yaml:11472]: maybe add more details on what the API mapping code is for

File: openapi.yaml [hunk 3]
```diff
@@ -156,6 +156,7 @@ paths:
          @@team_name1@@ | Resolves to the recipient’s ‘Team Name’ field.<br/><br/> For example, Red Team. 
          @@messagelabel@@| Resolves to the label field of the sent message. <br/><br/>For example, Incident #1234. 
          @@messagecategories@@ | Resolves to the list of categories used in the message, separated with commas. <br/><br/>For example, Product Update. 
+         @@web_link@@ | Resolves to a generated unsuubscribe link. <br /><br />For example, <pre>https://<region>.whispir.com/unsubscribe?_parameters=value</pre>
```
Comment (lnachor-whispir) [inline openapi.yaml:159]: typo in unsubscribe word

Comment (lnachor-whispir) [reply openapi.yaml:159]: also, @@web_link@@ is not always for unsubscribe - it is generally used as a link to the web channel content.  We just reused it for unsubscribe

**Review by Fatal-Errol — APPROVED** [review]

File: openapi.yaml [hunk 4]
```diff
@@ -463,6 +464,77 @@ paths:
...
+        <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+        <ns2:message xmlns:ns2="http://schemas.api.whispir.com" xmlns:ns3="http://schemas.api.whispir.com/dap">
+          <to>john.smith@test.com</to>
+          <subject>Test scheduled email message</subject>
+          <email>
+              <body>This is my scheduled content</body>
+              <type>text/plain</type>
+              <optOutOption>
+                <rmsUnsubscribeUrl>@@web_link@@#unsubscribe</rmsUnsubscribeUrl>
+                <oneClickApiMapping>unsubscribe</oneClickApiMapping>
```
Comment (lnachor-whispir) [inline openapi.yaml:518]: pls change to a sample generated api mapping code

File: openapi.yaml [hunk 5]
```diff
@@ -463,6 +464,77 @@ paths:
...
+        ```
+
+        In this example we're utilizing the `@@web_link@@` tag that generates an unsubscribe link for the recipients.
+        The `web` payload is required when using the `@@web_link@@` value in the `email`.`optOutOption`.
+        
+        The `rmsUnsubcribeUrl` is a URL link to the web channel unsubscribe content for `GET` requests, and for one-click unsubscribe functionality via `POST` requests.
+
+        The `oneClickApiMapping` property requires you to have an existing API Mapping and have access to its details. This API Mapping is triggered for one-click unsubscribe functionality.
+        
+        Both the `rmsUnsubscribeUrl` and `oneClickApiMapping` properties are optional.
```
Comment (lnachor-whispir) [inline openapi.yaml:537]: both are optional, but rmsUnsubscribeUrl is required to enable unsubscribe and oneClickApiMapping is required to enable one-click unsubscribe

**Review by lnachor-whispir — APPROVED** [review]

File: openapi.yaml (hunk 1 above)
Comment (aosmawhispir) [reply openapi.yaml]: I'll resolve this comment :) @lnachor-whispir

File: openapi.yaml (hunk 2 above)
Comment (aosmawhispir) [reply openapi.yaml]: I'll resolve this comment :) @lnachor-whispir

File: openapi.yaml [hunk 6]
```diff
@@ -11394,6 +11451,25 @@ components:
...
+      x-examples:
+        rmsUnsubscribeUrl: '@@web_link@@'
+        oneClickApiMapping: "unsubscribe"
+      x-tags:
+        - Email
+      properties:
+        rmsUnsubscribeUrl:
+          type: string
+          example: "@@web_link@@#unsubscribe"
+          description: This is the unsubscribe URL of your message.
```
[the remark this answers is not in the document]
Comment (aosmawhispir) [reply openapi.yaml]: I'll resolve this comment :) @lnachor-whispir

File: openapi.yaml (hunk 3 above)
Comment (aosmawhispir) [reply openapi.yaml]: I'll resolve this comment :) @lnachor-whispir

File: openapi.yaml (hunk 4 above)
Comment (aosmawhispir) [reply openapi.yaml]: I'll resolve this comment :) @lnachor-whispir

File: openapi.yaml (hunk 5 above)
Comment (aosmawhispir) [reply openapi.yaml]: I'll resolve this comment :) @lnachor-whispir

**Pull request merged**